### DISCUSSION TITLE
Keycloak default schedulers time configuration

### DISCUSSION THREAD

**Discussion by Rangadhama** [keycloak/keycloak#21864] [Q&A]
Hi Team,

Greetings !!

We have enabled the User Events in Keycloak , which logs to **EVENT_ENTITY** table and we have set the expiration time to clear from table , **it says once expiration is met , records will be deleted periodically**
![image](https://github.com/keycloak/keycloak/assets/49231664/5386bb65-d83a-4d19-9b35-17f4ec7c2d6e)

And in one of forums , we got periodical run is every 15 mins and it will delete the records.
We need to know where we can update this time ( 15 mins) to different value and run the schedulers on preferred time.

Please help and guide us on where we can get this configuration and how to update the same value.

Thanks in Advance !!

**Comment by danielFesenmeyer** [comment]
~~The key for the corresponding config is `scheduled.interval`.~~ **EDIT**: See my comment below

See https://github.com/keycloak/keycloak/blob/bd0f87fc4d281b869ef067afc6bfd8f5eaf3848b/model/legacy-private/src/main/java/org/keycloak/storage/datastore/LegacyDatastoreProviderFactory.java#L101
The value should be an int in **seconds**.

For example, you could define that in Keycloak's quarkus.properties - or any other way which is supported by Microprofile (e.g. env var).

**Reply by Rangadhama** [reply]
Thanks for the reply !

Configuration is not working 
Please find below actions done 

Keycloak version : 19.0.2
Added  file named **quarkus.properties** into the **/opt/keycloak/conf/** folder 
and added property **scheduled.interval** = 60000 ## 10 hours interval

Still the Events_Entity table cleared around 15 mins after expiration is met for the records created by default keycloak scheduler.
If there is any ENV variable available for this configuaration please let us know

Thanks in Advance !!

**Reply by danielFesenmeyer** [reply]
Sorry, I got that wrong.

You should be able to define the following value in keycloak.conf:
`spi-scheduled-interval=60000`

Alternatively, you can also use a CLI arg or Env as described for the settings here: https://www.keycloak.org/server/all-provider-config
